### PR TITLE
💄 Replace lang tags with flag emoji

### PR DESCRIPTION
**Pull request by foysalit**
This PR replaces lang tags with the associated flag emoji, if there is one.

Screenshot 2024-10-03 at 21 40 00

> in quick action panel

Screenshot 2024-10-03 at 21 40 29

> in workspace

### CODE REVIEW

File: lib/locale/languages.ts [hunk 1]
```diff
@@ -2,545 +2,2612 @@ interface Language {
...
-  { code3: 'alb', code2: 'sq', name: 'Albanian' },
-  { code3: 'ale', code2: '', name: 'Aleut' },
-  { code3: 'alg', code2: '', name: 'Algonquian languages' },
-  { code3: 'alt', code2: '', name: 'Southern Altai' },
-  { code3: 'amh', code2: 'am', name: 'Amharic' },
-  { code3: 'ang', code2: '', name: 'English, Old (ca.450-1100)' },
-  { code3: 'anp ', code2: 'Angika', name: 'Angika' },
-  { code3: 'apa', code2: '', name: 'Apache languages' },
-  { code3: 'ara', code2: 'ar', name: 'Arabic' },
+  {
```
Comment (matthieusieben) [inline lib/locale/languages.ts:9]: nit: Wouldn't this be more readable using 1 line per item?

Comment (matthieusieben) [reply lib/locale/languages.ts:9]: Also, would 'this be more memory efficient by omitting `code2` if empty?

**Review by matthieusieben — APPROVED** [review]
looks good. Any idea why the build failed ?

https://github.com/bluesky-social/ozone/actions/runs/11187349902/job/31104207001?pr=211

File: lib/locale/languages.ts (hunk 1 above)
Comment (foysalit) [reply lib/locale/languages.ts:9]: prettier automatically formatted these but also, once compiled, I don't think it matters much, right?
cod3 is used for a different feature I think, can't recall exactly so we kinda need every item here.

**Comment by foysalit** [comment]
> looks good. Any idea why the build failed ?
> 
> https://github.com/bluesky-social/ozone/actions/runs/11187349902/job/31104207001?pr=211

yeah there was a type error when main was merged in. fixed the issue and made build pass before merging.

**Comment by mdruker** [comment]
Sorry for the unsolicited comment but as an FYI associated flags for some languages can be very politically heated, e.g. a lot of people who use the Russian language who would be offended at an association with the flag of Russia

**Comment by IsaacMarovitz** [comment]
I'm not sure if I like this change from a UX perspective either.

**Comment by IsaacMarovitz** [comment]
> Sorry for the unsolicited comment but as an FYI associated flags for some languages can be very politically heated, e.g. a lot of people who use the Russian language who would be offended at an association with the flag of Russia

This is an excellent point. Looking through the changeset I can see *several* potentially politically charged changes here.

**Pull request merged**